### PR TITLE
Monoisotopic masses can be calculated correctly.

### PR DESCRIPTION
**Pull request by janlisec**
Suggested Version: 1.65.4
Changes:
    * Rdisop-1.65.4
    * !changed `getIsotope` function to return a list of matrices to be consistent with the other getter functions
    * new exported function `getMono()` which will return the mono-isotopic mass for a molecule object (also when a list of molecules as for decomposeMass) or alternatively for a vector of chemical formulas
    * modified documentation to clarify the difference of exactmass and mono-isotopic mass
    * `testthat` test for `getMono` were set up 
    * new internal function to count chemical elements (imported/copied from the CorMID package) to be utilized in `getMono`
    * two data sets included in package and appropriately documented:
    * `isotopes` (a data.frame version of the PSE info) 
    * `mono_masses` (a named vector containing only the most abundant isotope per element)

### CODE REVIEW

**Comment by janlisec** [comment]
Sorry, Steffen. The PR passed my local RCHECK and the GitHub RCHECK in [my fork](https://github.com/janlisec/Rdisop). I maybe forgot to include the `LazyData: true` statement in the Description, which is why one testthat tests for the new `getMono` function fails. In this function I load package data by `Rdisop::mono_masses`, and this statement fails. Can you trouble shoot this before accepting the PR?

**Comment by janlisec** [comment]
@sneumann I managed to amend the PR by the missing LazyData statement. Now all checks are passed. If you are fine with the changes, please accept, so I can take care of the next issue from the repository's list.

File: NAMESPACE [hunk 1]
```diff
@@ -8,6 +8,7 @@ export(getFormula)
 export(getIsotope)
 export(getMass)
 export(getMolecule)
+export(getMono)
```
Comment (sneumann) [inline NAMESPACE]: Can I have `getMonoisotopic` ? The abbreviation looses meaning without context

File: R/CountChemicalElements.R [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#' @details No testing for any chemical alphabet is performed. Elements may occur
+#'  several times and will be summed up in this case without a warning.
+#'  Information within brackets, i.e. [13]C will be removed prior to counting
+#'  together with other symbols not contained in the R set 'alnum'.
+#'  The result is filtered and ordered according to parameter 'ele' if provided.
+#' @param x Chemical formula.
+#' @param ele Character vector of elements to count particularly or counting all contained in string if NULL.
+#' @return A named numeric with counts for all contained or specified elements.
+#' @references This function is a copy from the `CorMID` package by Jan Lisec.
+#'   It is not (and should bot be) exported in Rdisop but only used internally.
```
Comment (sneumann) [inline R/CountChemicalElements.R:13]: What about starting internal function with a dot, `.CountChemicalElements()`

File: R/getMolecule.R [hunk 3]
```diff
@@ -3,25 +3,29 @@
 #' @aliases getMass
 #' 
 #' @description Parse the sum formula and calculate the theoretical exact mass 
-#'     and the isotope distribution.
+#'     and the isotope distribution for an approximate MS resolution of 20,000.
```
Comment (sneumann) [inline R/getMolecule.R]: Does the 20.000 come from the Boecker paper ? For sure there is no isotope fine structure. 
Maybe mention that instead ?

File: R/getMolecule.R [hunk 4]
```diff
@@ -3,25 +3,29 @@
...
 #' @param z Charge z of molecule for exact mass calculation.
 #' @param maxisotopes Maximum number of isotopes shown for the resulting molecule.
 #'
 #' @details \code{getMolecule()} will parse the sum formula and calculate the 
-#'     theoretical exact monoisotopic mass and the isotope distribution. For a 
-#'     given element, return the different mass values.
+#'     exact mass and the isotope distribution.
+#'     The exact mass is the mass of the most abundant isotope and is not 
+#'     identical with the monoisotopic mass. The latter can be extracted using 
+#'     the function `getMono()`. This function can also be supplied with a
```
Comment (sneumann) [inline R/getMolecule.R]: if `getMono()` is changed, then here as well.

**Review by sneumann — COMMENTED** [review]
Hi, thanks for the PR. I just have tiny suggestions. Yours, Steffen

File: R/getMolecule.R (hunk 3 above)
Comment (janlisec) [reply R/getMolecule.R]: In `enviPat` it is possible to adjust the resolution. By comparing the results with `Rdisop` I estimated what is probably internally used. I feel it is important for the user to get an approximate idea of the resolution he can expect. However, if you feel uncomfortable to provide this number, feel free to change to 'no isotope fine structure' instead.

File: NAMESPACE (hunk 1 above)
Comment (janlisec) [reply NAMESPACE]: Sure.

File: R/CountChemicalElements.R (hunk 2 above)
Comment (janlisec) [reply R/CountChemicalElements.R:13]: Fine with me.

File: R/getMolecule.R (hunk 4 above)
Comment (janlisec) [reply R/getMolecule.R]: Yes. There are more instances (e.g. the Changes-document and probably a testthat file). Please Search and replace the whole package.

**Comment by janlisec** [comment]
@sneumann I am not sure about how we proceed. :/ Will you make the changes while accepting the PR or shall I open another PR where the modifications are incorporated? Bests, Jan

**Comment by sneumann** [comment]
Hi, no need for a separate PR. You can go ahead and fix/rename things, push to your [janlisec:devel](https://github.com/janlisec/Rdisop/tree/devel) and these changed automagically go into this PR here. Worst case ping me again, and I can merge all in one go. Yours, Steffen

**Comment by janlisec** [comment]
@sneumann Worked as suggested. :) I smuggled an additional modification in which adresses issue #3. Basically I use `.CountChemicalElements` to pre-check a sum formula submitted to `getMolecule` to avoid C++ crashes occuring in case of syntactically incorrect formulas. Bests, Jan

**Comment by sneumann** [comment]
Merged & Pushed as 1.67.1 to https://bioconductor.org/checkResults/devel/bioc-LATEST/Rdisop/
Thanks Jan ! Yours, Steffen

**Comment by sneumann** [comment]
Hm, maybe my manual changes prevented GitHub from seeing it was merged in e70ef183b3ce8042b9f8ac6f37b3957c8f924617

**Pull request closed without merging**